### DISCUSSION TITLE
Is there any documentation about how to use `absl::ParseCommandLine`?

### DISCUSSION THREAD

**Discussion by zuoxinyu** [abseil/abseil-cpp#1447] [General]
Any docs can tell me that I should link against `absl::flags_parse`？Why are there soooo many `libabsl_xxx.a`？

**Comment by derekmauro** [comment]
https://abseil.io/docs/cpp/guides/flags

If you are calling `absl::ParseCommandLine()` then you need to link against it. Our primary build system is Bazel, and the design of Bazel encourages this design with many libraries.

**Reply by zuoxinyu** [reply]
> https://abseil.io/docs/cpp/guides/flags

No, this documentation does not even mention the linking directive (or convention?) for Bazel. The official website says nothing about how to link against these ~~stupid~~ trivial libraries, a simple `hellow_world` example doesn`t help at all. Please add a description for more detailed link rules for CMake users.
~~Otherwise maybe you could remove the CMake support entirely.~~

> and the design of Bazel encourages this design with many libraries.

This results in a loooong build command, for example, I only want 2 APIs in abseil, `ABSL_FLAG` and `absl:ParseCommandLine`, but the build tool generates clang cmd like:

<details><summary>Details</summary>
<p>

/usr/bin/clang++ ... -labsl_bad_any_cast_impl -labsl_cordz_sample_token -labsl_die_if_null -labsl_failure_signal_handler -labsl_flags_parse -labsl_flags_usage -labsl_flags_usage_internal -labsl_leak_check -labsl_log_flags -labsl_log_initialize -labsl_log_internal_check_op -labsl_flags -labsl_flags_internal -labsl_flags_marshalling -labsl_flags_reflection -labsl_flags_private_handle_accessor -labsl_flags_commandlineflag -labsl_flags_commandlineflag_internal -labsl_flags_config -labsl_flags_program_name -labsl_log_internal_conditions -labsl_log_internal_message -labsl_examine_stack -labsl_log_internal_format -labsl_log_internal_proto -labsl_log_internal_nullguard -labsl_log_internal_log_sink_set -labsl_log_internal_globals -labsl_log_globals -labsl_log_sink -labsl_log_entry -labsl_hash -labsl_city -labsl_low_level_hash -labsl_periodic_sampler -labsl_random_internal_distribution_test_util -labsl_random_distributions -labsl_random_seed_sequences -labsl_random_internal_pool_urbg -labsl_random_internal_randen -labsl_random_internal_randen_hwaes -labsl_random_internal_randen_hwaes_impl -labsl_random_internal_randen_slow -labsl_random_internal_platform -labsl_random_internal_seed_material -labsl_random_seed_gen_exception -labsl_raw_hash_set -labsl_hashtablez_sampler -labsl_scoped_set_env -labsl_statusor -labsl_status -labsl_cord -labsl_cordz_info -labsl_cord_internal -labsl_cordz_functions -labsl_exponential_biased -labsl_cordz_handle -labsl_crc_cord_state -labsl_crc32c -labsl_crc_internal -labsl_crc_cpu_detect -labsl_bad_optional_access -labsl_str_format_internal -labsl_strerror -labsl_synchronization -labsl_graphcycles_internal -labsl_stacktrace -labsl_symbolize -labsl_debugging_internal -labsl_demangle_internal -labsl_malloc_internal -labsl_time -labsl_civil_time -labsl_strings -labsl_strings_internal -labsl_base -labsl_spinlock_wait -labsl_int128 -labsl_throw_delegate -labsl_time_zone -labsl_bad_variant_access -labsl_raw_logging_internal -labsl_log_severity

</p>
</details> 

This is very annoying to read for troubleshooting.

**Comment by tituswinters** [comment]
Don't be rude. Try that again without insulting the people you are asking
to help you.

**Reply by zuoxinyu** [reply]
Sorry for the use of vulgar words. I chose wrong tag for this thread. I'm just complaining about the unfriendly documentation.
